### PR TITLE
Use functions in http interceptors not pointers

### PR DESCRIPTION
**Pull request by riyafa**
## Automation tests
 - Integration tests
   > Existing tests were modified

### CODE REVIEW

File: stdlib/ballerina-http/src/main/ballerina/http/authn_filter.bal [hunk 1]
```diff
@@ -26,54 +26,45 @@ AuthnHandlerChain authnHandlerChain = new(registry);
 @Field {value:"filterRequest: request filter method which attempts to authenticated the request"}
 @Field {value:"filterRequest: response filter method (not used this scenario)"}
 public type AuthnFilter object {
-    public {
-        function (Request request, FilterContext context) returns (FilterResult) filterRequest;
-        function (Response response, FilterContext context) returns (FilterResult) filterResponse;
+    @Description {value:"filterRequest: Request filter function"}
+    public function requestFilter (Request request, FilterContext context) returns FilterResult {
+        boolean authenticated;
```
Comment (isurulucky) [inline stdlib/ballerina-http/src/main/ballerina/http/authn_filter.bal]: Formatting required.

File: stdlib/ballerina-http/src/main/ballerina/http/authn_filter.bal [hunk 2]
```diff
@@ -26,54 +26,45 @@ AuthnHandlerChain authnHandlerChain = new(registry);
...
+				}
+				() => {
+					authenticated = authnHandlerChain.handle(request);
+				}
+			}
+		} else {
+			// not secured
+			return createAuthnResult(true);
+		}
+		return createAuthnResult(authenticated);
```
Comment (isurulucky) [inline stdlib/ballerina-http/src/main/ballerina/http/authn_filter.bal]: Format the code.

File: distribution/docker/dependency-reduced-pom.xml [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    </dependency>
+  </dependencies>
+  <properties>
+    <maven.findbugsplugin.version.exclude>findbugs-exclude.xml</maven.findbugsplugin.version.exclude>
+    <jackson.version>2.6.1</jackson.version>
+    <surefireArgLine>-Dfile.encoding=UTF-8</surefireArgLine>
+    <mavan.findbugsplugin.exclude.file>findbugs-exclude.xml</mavan.findbugsplugin.exclude.file>
+  </properties>
+</project>
+
```
Comment (isurulucky) [inline distribution/docker/dependency-reduced-pom.xml]: Is this file required? This is not relevant for filters.

File: stdlib/ballerina-http/src/main/ballerina/http/authz_filter.bal [hunk 4]
```diff
@@ -28,55 +28,47 @@ HttpAuthzHandler authzHandler = new(authzCache);
...
+	@Param {value:"context: FilterContext instance"}
+	@Return {value:"FilterResult: Authorization result to indicate if the request can proceed or not"}
+    public function requestFilter (Request request, FilterContext context) returns FilterResult {
+		// first check if the resource is marked to be authenticated. If not, no need to authorize.
+		// TODO: check if we can remove this once the security context is there.
+		//if (!isResourceSecured(context)) {
+		//    // let the request pass
+		//    return createAuthzResult(true);
+		//}
+		// check if this resource is protected
```
Comment (shafreenAnfar) [inline stdlib/ballerina-http/src/main/ballerina/http/authz_filter.bal]: Do we need this commented code ?

File: stdlib/ballerina-http/src/main/ballerina/http/authz_filter.bal [hunk 5]
```diff
@@ -28,55 +28,42 @@ HttpAuthzHandler authzHandler = new(authzCache);
...
-                authorized = false;
-            }
-        }
-        () => {
-            // scopes are not defined, no need to authorize
-            authorized = true;
-        }
+    @Description {value:"Initializes the AuthzFilter"}
+    public function init (){
+        authzHandlerChain = createAuthzHandlerChain();
```
Comment (isurulucky) [inline stdlib/ballerina-http/src/main/ballerina/http/authz_filter.bal]: This should be removed. There is no authzHandlerChain variable now.

**Comment by riyafa** [comment]
Another PR will be sent on top of this
